lib/package.js

```diff
@@ -509,6 +509,12 @@
 
   function loadPackageConfigPath(loader, pkgName, pkgConfigPath) {
     var configLoader = loader.pluginLoader || loader;
+    if (configLoader.execute === false) {
+      var config = configLoader.getConfig();
+      configLoader = new configLoader.constructor();
+      configLoader.config(config);
+      configLoader.execute = true;
+    }
```

Comment:
We may as well cache this as `loader.pluginLoader` though so we don't have to generate the loader and config each config load.